vpnglean/glean_parser_ext/templates/cpp.jinja2

```diff
@@ -38,28 +38,34 @@ struct {{ obj.name|Camelize }}{{ suffix }} {
   {% endfor %}
 
   FfiExtra ToFfiExtra() const {
-    QVector<const char*> extraKeys;
-    QVector<const char*> extraValues;
+    const char* extraKeys[{{ obj|attr(name)|length }} + 1];
+    const char* extraValues[{{ obj|attr(name)|length }} + 1];
+    int count = 0;
+
+    QVector<QByteArray> _keepStringsAlive;
     {% for item, type in obj|attr(name) %}
     if (_{{item|camelize}}.canConvert<{{type|extra_type_name}}>()) {
-      extraKeys.append("{{item}}");
+      extraKeys[count] = "{{item}}";
       {% if type == "string" %}
-      extraValues.append(_{{item|camelize}}.toString().toUtf8().constData());
+      _keepStringsAlive.append(_{{item|camelize}}.toString().toUtf8());
+      extraValues[count] = _keepStringsAlive[count].constData();
       {% elif type == "boolean" %}
-      extraValues.append(_{{item|camelize}}.toBool() ? "true" : "false");
+      _keepStringsAlive.append(_{{item|camelize}}.toBool() ? "true" : "false");
```

Comment:
because these are "static" strings, you don't need to keep them alive. YOu can directly do: `extraValues[count] = _{{item|camelize}}.toBool() ? "true" : "false"`

Reply:
Duh, of course! Thanks for catching :)

Reply:
Uh oh, actually I can't do this. I had this because I wanted to keep the index of keepStringsAlive in sync with the index of the `extraValues`.

Reply:
I added a comment explaining this.

Reply:
Hm, but this is going to fail since it is not a QByteArray right. I need to rethink this. The only reason why it did not fail is because we don't have boolean type extras right now.

Reply:
Ok yeah now this is addressed. Please see: [9b9e861](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5047/commits/9b9e861d38206440019bab1b9fe2223a9167952c)